src/config.js

```diff
@@ -17,6 +17,35 @@ type ApplyConfigToArgvParams = {|
   configFileName: string,
 |};
 
+type ApplySubOptionsToArgvParams = {|
+  commandExecuted: string,
+  argv: Object,
+  mapCommandToSubOpts: Object,
+  configObject: Object,
+  defaultValues: Object,
+  configFileName: string,
+|};
+
+export function applySubOptionsToArgv({
+  commandExecuted,
+  argv,
+  mapCommandToSubOpts,
+  configObject,
+  defaultValues,
+  configFileName,
+}: ApplySubOptionsToArgvParams): Object {
+  // cannot use spread operator because Flow complains. See: https://github.com/facebook/flow/pull/3381/
+  const newArgv = JSON.parse(JSON.stringify(argv));
+  if (mapCommandToSubOpts[commandExecuted]) {
+    for (const opt in mapCommandToSubOpts) {
+      newArgv[opt] = mapCommandToSubOpts[opt];
```

Comment:
`mapCommandToSubOpts` [looks like](https://github.com/mozilla/web-ext/pull/795/files#diff-9589e3a0cc72e26a4814bc9ef05aa2d4R74) it will only contain option definitions. I don't think you should be assigning those to `newArgv`. You need to assign values from `configObject` only.